### ISSUE TITLE
spread orders always rejected

### ISSUE THREAD

**Issue by klepsydra** [jmfernandes/robin_stocks#105]
Could you shed light on why all my option spread orders placed using order_option_spread keep getting rejected? If placed far outside market hours, not right away, but eventually an email at most says 

> Your limit order to close 1 FITB Put Credit Spread was rejected by an exchange on April 14th, 2020 at 10:46 AM. 

but never why.

All my _close_ vs. _open_ effect, _buy_ vs. _sell_ action for each leg, quantity etc. seem to line up with what you'd expect, else the Robinhood API immediately returns an appropriate error.

I carefully followed
https://github.com/jmfernandes/robin_stocks/blob/master/examples/place_vertical_spread.py
and by its example, tried putting quotation marks around the strike prices of each of the legs in the _spread_ dict argument, but it makes no difference whether number or string.

This is my method call:
`order_option_spread(direction="credit",price=0.27,symbol="FITB",quantity=1,spread=[{"expirationDate":"2020-05-15","strike":"15.00","optionType":"put","action":"sell","effect":"close"},{"expirationDate":"2020-05-15","strike":"16.00","optionType":"put","action":"buy","effect":"close"}],timeInForce="gtc")`

Resulting in output:

```
Found Additional pages.
Loading page 2 ...
Found Additional pages.
Loading page 2 ...
{ 'cancel_url': 'https://api.robinhood.com/options/orders/07da4e58-9f26-4ef1-8c26-547318645f04/cancel/',
  'canceled_quantity': '0.00000',
  'chain_id': 'bc6be536-d34d-419a-9285-bb3ef3ac6c83',
  'chain_symbol': 'FITB',
  'closing_strategy': 'short_put_spread',
  'created_at': '2020-04-14T15:03:10.208709Z',
  'direction': 'credit',
  'id': '07da4e58-9f26-4ef1-8c26-547318645f04',
  'legs': [ { 'executions': [],
              'id': '6bb6dc33-929e-4a08-b64b-ec4262dc3f71',
              'option': 'https://api.robinhood.com/options/instruments/26495dc1-1f85-4010-a282-845d0fe6f4f5/',
              'position_effect': 'close',
              'ratio_quantity': 1,
              'side': 'buy'},
            { 'executions': [],
              'id': '20700688-19a9-40ad-8939-37b37c850b5e',
              'option': 'https://api.robinhood.com/options/instruments/e9f32564-b6a8-47c3-8843-0fc37d724f02/',
              'position_effect': 'close',
              'ratio_quantity': 1,
              'side': 'sell'}],
  'opening_strategy': None,
  'pending_quantity': '1.00000',
  'premium': '27.00000000',
  'price': '0.27000000',
  'processed_premium': '0.0000',
  'processed_quantity': '0.00000',
  'quantity': '1.00000',
  'ref_id': '4b1a3000-e4be-4a39-badd-64d42bdd36bb',
  'response_category': None,
  'state': 'unconfirmed',
  'stop_price': None,
  'time_in_force': 'gtc',
  'trigger': 'immediate',
  'type': 'limit',
  'updated_at': '2020-04-14T15:03:10.208734Z'}

```

Which itself looks intact.  Notice initially the  'state' == 'unconfirmed'

After about a minute it gets rejected, when placed during or nearby market hours.

If I place the same exact order to close the option spread using the app, i.e. to close this credit spread at debit price of 0.27, it just works, showing a Queued state, then Placed.

I've tried things like rounding strikes and prices to two decimal places, swapping buy and sell or even setting both to the same action, etc.

Stock and single option orders placed via robin_stocks seems to get accepted fine, meanwhile.

By the way, the documentation for order_option_spread is missing some mandatory fields only revealed in the source:  https://robin-stocks.readthedocs.io/en/latest/functions.html#robin_stocks.orders.order_option_spread 
The spread (dict) should have more values than shown: "action" and "effect" are missing.

**Comment by klepsydra** [comment]
I may have figured it out...pending tests next market session.

No thanks to the Robinhood app, whose on-screen order status has been painfully (and expensively) misleading.  Namely, when showing an order to close a vertical option spread, both legs are inappropriately marked as either Buy or Sell, when by definition they must always be opposites. Eg. see how both:
 

https://lh3.googleusercontent.com/848MYNhJ8qlGim6drRzWkNurKFAJz-u8g4MEmPc0MpojbO0V2A7I4XyWdsh9Rbj0EHYH9B-7bRzHFdjYdV8S7OiKJaii6fuM4cpxCxmI79hTcRy9VcQCsOVQSXYChIYGQkV9EcvnoW7ZbwiMYnL8GU6m8569AEMidnFEMsDGs-hz4oPsGZeIOaohvceOVGkX4R-dZVG2N82aN8zqbwhoA20jbWY7pS4CYHg90zDxEPFgSwgWKBINhpqDAxSqknSvJrw7yT3lqgqoVuEey_F68VsSaE86Ujr6fxjju-trcj8NYlk5GaMRovllHHzGYb-Pg4ph-zi_L3bbGiBHqoaZNyG3_2ewCVrQy_a6uVkTFWVTg9zdIfkiDo_aqOGb6yyHSXMO5hJEqYqBL43erdC9eT_aZJ2lITHroj5HxBB7Ovz2JrSbSD11GY7gx8noyq503Q4RMfo3zRP_7o0lZAEvY_QjjywUHDANLPNocezmdQmnbHoM8fik9unHtjIEBTk9nVYBrnr6jpF_SRYD8ks-oNqOPlF2B_WS6vWir1FMtOvN3ZmYJ0Fx0Hub3nqa4jvNd6FltxolIXdRqQjBJTB3W6OTxYU2Qi83_vYYAaFMqR_PxWcaxQkI83_DdfdIYOr1AJscJ0lKtycf57kVTCITcoZItKDDVrdVldrr5ED5a5HGb3P-nI734TmFmSMzVg=w308-h547-no


Basically, the "direction" has to reverse when you try to close a spread position.

I.e. 
`direction="credit"`
when
`"action":"sell","effect":"close"`

in the case of a long call or put spread.

**Comment by jmfernandes** [comment]
Thanks a lot for looking at this. Yeah I noticed this too when placing and closing spreads in the mobile app. Clicking on the pending order says I'm selling both legs which made me panic a little the first time I saw it. 

Obviously robinhood isn't submitting that information to the stock market because it's wrong. They must have built some logic into their system behind the scenes which calculates the correct information from these generic buy and sell settings they've assigned to the spreads. 

It sounds like you are close to figuring out robinhoods rules for what they expect as inputs. If you can figure it out I can add it to the code to fix it.

The Joy's of private api's........

**Comment by klepsydra** [comment]
Actually, it's worse because they DO appear to blindly pass them through to
the exchanges.

Twice when I attempted to auto-close long VXX put spreads,
direction=debit (like in the original position) and effect=close,
they got filled at 0 price, instead of crediting me what I input.

Basically I lost positions of positive value for free.

Instead of getting rejected, like most of my earlier trials, support
confirmed that they filled at 0 due to "price improvement", otherwise I
would have actually paid again to close those long spreads.  Ouch.

It makes sense that way outside market hours, option orders merely get
queued at this broker, and not seen by an exchange/market maker until
premarket.
So the exchanges were likely the ones with courteous failsafes rejecting
orders back to the broker, when the price differences (from current market
values) are more extreme.  My lost VIX spreads went for under $10.

**Comment by klepsydra** [comment]
Reason for failure identified in: https://github.com/jmfernandes/robin_stocks/issues/105#issuecomment-623242472
